gateserver/controller_api.py

```diff
@@ -0,0 +1,48 @@
+from .controller_protocol import (checkmsg, MsgType, ResponseStatus, Response,
+    response_packet_for, parse_packet, request_from_packet, BadMessageError)
+from . import db
+from . import utils
+
+def get_key_for_mac(mac):
+    """Loads the key for this controller from the DB."""
+    rs = db.exec_sql('SELECT key FROM controller WHERE id = %s',
+                     (utils.bytes2mac(mac),), ret=True)
+    checkmsg(len(rs) == 1, 'unknown controllerID')
+    return bytes(rs[0]['key'])
+
+def isic_id_repr(buf):
+    """See https://github.com/fmfi-svt-gate/server/wiki/Controller-%E2%86%94-Server-Protocol#note-isic-ids-representation"""
+    length, string = buf[0], buf[1:]
+    checkmsg(length <= len(string), 'isic_id_repr length > buffer size')
+    return string[:length]
+
+def log_message(controllerID, mtype, indata, status):
```

Comment:
`log_message` sa mení v nasledujúcom (ešte nepushnutom, lebo ešte neupdatnutom) PR, preto som to nechytala. Ale spravím.